genesyscloud/resource_genesyscloud_knowledge_category.go

```diff
@@ -0,0 +1,281 @@
+			if getErr != nil {
+				return nil, diag.Errorf("Failed to get page of knowledge categories: %v", getErr)
+			}
+
+			if knowledgeCategories.Entities == nil || len(*knowledgeCategories.Entities) == 0 {
+				break
+			}
+
+			for _, knowledgeCategory := range *knowledgeCategories.Entities {
+				id := fmt.Sprintf("%s %s %s", *knowledgeCategory.Id, *knowledgeCategory.KnowledgeBase.Id, *knowledgeCategory.LanguageCode)
+				resources[id] = &ResourceMeta{Name: *knowledgeCategory.Name}
+			}
+		}
+	}
+
+	return resources, nil
+}
+
+func knowledgeCategoryExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllKnowledgeCategories),
+		RefAttrs:         map[string]*RefAttrSettings{}, // No references
+	}
+}
+
+func resourceKnowledgeCategory() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Knowledge Category",
+
+		CreateContext: createWithPooledClient(createKnowledgeCategory),
+		ReadContext:   readWithPooledClient(readKnowledgeCategory),
+		UpdateContext: updateWithPooledClient(updateKnowledgeCategory),
+		DeleteContext: deleteWithPooledClient(deleteKnowledgeCategory),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"knowledge_base_id": {
+				Description: "Knowledge base id of the category",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"language_code": {
+				Description:  "language code of the category",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"en-US", "en-UK", "en-AU", "de-DE", "es-US", "es-ES", "fr-FR", "pt-BR", "nl-NL", "it-IT", "fr-CA"}, false),
+			},
+			"knowledge_category": {
+				Description: "Knowledge category parent id",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Required:    true,
+				Elem:        knowledgeCategory,
+			},
+		},
+	}
+}
+
+func createKnowledgeCategory(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	languageCode := d.Get("language_code").(string)
+	knowledgeBaseId := d.Get("knowledge_base_id").(string)
+	knowledgeCategory := d.Get("knowledge_category").([]interface{})[0].(map[string]interface{})
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	knowledgeAPI := platformclientv2.NewKnowledgeApiWithConfig(sdkConfig)
+
+	knowledgeCategoryRequest := buildKnowledgeCategory(knowledgeCategory)
+
+	log.Printf("Creating knowledge category %s", knowledgeCategory["name"].(string))
+	knowledgeCategoryResponse, _, err := knowledgeAPI.PostKnowledgeKnowledgebaseLanguageCategories(knowledgeBaseId, languageCode, knowledgeCategoryRequest)
+	if err != nil {
+		fmt.Errorf("Failed to create knowledge category %s: %s", *knowledgeCategoryResponse.Name, err)
+		return diag.Errorf("Failed to create knowledge category %s: %s", *knowledgeCategoryResponse.Name, err)
```

Comment:
If the post isn't successful, I don't think `*knowledgeCategoryResponse.Name` will have any value in it, so it's better to use `knowledgeCategory["name"].(string)`. Or maybe just leave it out since name is an optional field so there mightn't be any value to begin with